src/managers/moduleManager.ts

```diff
@@ -0,0 +1,57 @@
+import { GnosisSafe } from '../../typechain'
+import { areAddressesEqual } from '../utils'
+import { SENTINEL_MODULES, zeroAddress } from '../utils/constants'
+
+class ModuleManager {
+  #ethers: any
+  #contract: GnosisSafe
+
+  constructor(ethers: any, contract: GnosisSafe) {
+    this.#ethers = ethers
+    this.#contract = contract
+  }
+
+  async getModules(): Promise<string[]> {
+    return this.#contract.getModules()
+  }
+
+  async isModuleEnabled(moduleAddress: string): Promise<boolean> {
+    return this.#contract.isModuleEnabled(moduleAddress)
+  }
+
+  async encodeEnableModuleData(moduleAddress: string): Promise<string> {
+    const isValidAddress = this.#ethers.utils.isAddress(moduleAddress)
+    if (!isValidAddress || moduleAddress === zeroAddress || moduleAddress === SENTINEL_MODULES) {
+      throw new Error('Invalid module address provided')
+    }
```

Comment:
Could be possibly extracted to a function for reusability and readability (as it will have a meaningful name)

Reply:
It's also used in ownerManager